projects/GKCore/GKCore/Cultures/PolishCulture.cs

```diff
@@ -28,5 +28,28 @@ public class PolishCulture : EuropeanCulture
         public PolishCulture()
         {
         }
+
+        public override string NormalizeSurname(string sn, bool aFemale)
+        {
+            if (string.IsNullOrEmpty(sn) || (sn[0] == '(' && sn[sn.Length - 1] == ')'))
+            {
+                sn = "?";
+            }
+            else
+            {
+                if (aFemale)
+                {
+                    sn = GetMaidenSurname(sn);
+
+                    if (sn.EndsWith("nа")) {
+                        sn = sn.Substring(0, sn.Length - 1) + "y";

```

Comment:
That's the most common case. There are other cases that are not cover by this, but it depends on the whole word rather than ending and would be hard or impossible to account for all possibilities